doc/manual/redirects.js

```diff
@@ -1,330 +1,413 @@
+    "ssec-relnotes-1.2": "release-notes/rl-1.2.html",
+    "ssec-relnotes-1.3": "release-notes/rl-1.3.html",
+    "ssec-relnotes-1.4": "release-notes/rl-1.4.html",
+    "ssec-relnotes-1.5.1": "release-notes/rl-1.5.1.html",
+    "ssec-relnotes-1.5.2": "release-notes/rl-1.5.2.html",
+    "ssec-relnotes-1.5": "release-notes/rl-1.5.html",
+    "ssec-relnotes-1.6.1": "release-notes/rl-1.6.1.html",
+    "ssec-relnotes-1.6.0": "release-notes/rl-1.6.html",
+    "ssec-relnotes-1.7": "release-notes/rl-1.7.html",
+    "ssec-relnotes-1.8": "release-notes/rl-1.8.html",
+    "ssec-relnotes-1.9": "release-notes/rl-1.9.html",
+    "ssec-relnotes-2.0": "release-notes/rl-2.0.html",
+    "ssec-relnotes-2.1": "release-notes/rl-2.1.html",
+    "ssec-relnotes-2.2": "release-notes/rl-2.2.html",
+    "ssec-relnotes-2.3": "release-notes/rl-2.3.html"
+  },
+  "language/values": {
+    "simple-values": "#primitives",
+    "lists": "#list",
+    "strings": "#string",
+    "lists": "#list",
+    "attribute-sets": "#attribute-set"
+  }
 };
 
-var isRoot = (document.location.pathname.endsWith('/') || document.location.pathname.endsWith('/index.html')) && path_to_root === '';
-if (isRoot && redirects[document.location.hash]) {
-  document.location.href = path_to_root + redirects[document.location.hash];
+
+function pathsMatch(a, b, path_to_root) {
+  // Do paths `a` and `b` match?
+  //
+  // This is more involved than it should be:
+  //
+  // 1. Path `b` can have an have an arbitrary prefix.
+  //
+  // 2. `path_to_root` consists only of `../`s and determines the depth
+  //     of `b` relative to the prefix:
+  //
+  //          `document.location.pathname`
+  //        |-----------------------------|
+  //        <prefix>/<path>/[<file>[.html]][#<anchor>]
+  //                 |----|
+  //             `path_to_root` has same number of segments
+  //
+  // 3. The following paths are equivalent:
+  //
+  //        foo/bar/
+  //        foo/bar/index.html
+  //        foo/bar/index
+  //
+  //  4. The following paths are also equivalent:
+  //
+  //         foo/bar/baz
+  //         foo/bar/baz.html
+  //
+  // We can use `path_to_root` to discern prefix from path.
+  //
+  // The last element of the following split is always empty.
+  // Example: '../../'.split('/') -> [ '..', '..', '' ]
+  const depth = path_to_root.split('/').length - 1;
+  var segmentsB = b.split('/');
+  // get file name of `b`
+  const fileB = segmentsB.pop(); // mutates `segmentsB`!
+  // get path of `b` without prefix and file name
+  const pathB = segmentsB.slice(segmentsB.length - depth).join('/');
+
+  var segmentsA = a.split('/');
+  const fileA = segmentsA.pop(); // mutates `segmentsA`!
+  const pathA = segmentsA.join('/')
+
+  function normalize(file) {
+    if (file === '') { return "index.html"; }
+    if (!file.endsWith('.html')) { return file + '.html'; }
+    return file;
+  }
+
+  return pathA === pathB && normalize(fileA) === normalize(fileB);
```

Comment:
We'd have to enforce that verbally, it's a vector for human error. Is there a reason for not keeping it that way?

Reply:
The main point is that it would allow us to drop the loop altogether as mentioned in the [review comment](https://github.com/NixOS/nix/pull/7006#pullrequestreview-1099037287). Not sure how much worth it is.

Reply:
Ah right, now I get it. Still unsure about enforcing correctness on developer side, but maybe this is irrelevant because it would rarely see change anyway.

I just imagine myself in a few months adding a new redirect and hating the setup for not telling me that it must be `foo/bar/index.html` instead of just `foo/bar/` as I see it in the URL.